frontend/apps/filemanager/filemanagerbookinfo.lua

```diff
@@ -0,0 +1,186 @@
+--[[--
+This module provides a way to display book information (filename and book metadata)
+]]
+
+local DocSettings = require("docsettings")
+local DocumentRegistry = require("document/documentregistry")
+local ImageViewer = require("ui/widget/imageviewer")
+local InfoMessage = require("ui/widget/infomessage")
+local InputContainer = require("ui/widget/container/inputcontainer")
+local KeyValuePage = require("ui/widget/keyvaluepage")
+local UIManager = require("ui/uimanager")
+local filemanagerutil = require("apps/filemanager/filemanagerutil")
+local lfs = require("libs/libkoreader-lfs")
+local util = require("util")
+local _ = require("gettext")
+
+local BookInfo = InputContainer:extend{
+    bookinfo_menu_title = _("Book information"),
+}
+
+function BookInfo:init()
+    if self.ui then -- only for Reader menu
+        self.ui.menu:registerToMainMenu(self)
+    end
+end
+
+function BookInfo:addToMainMenu(menu_items)
+    menu_items.book_info = {
+        text = self.bookinfo_menu_title,
+        callback = function()
+            -- Get them directly from ReaderUI's doc_settings
+            local doc_props = self.ui.doc_settings:readSetting("doc_props")
+            -- Make a copy, so we don't add "pages" to the original doc_props
+            -- that will be saved at some point by ReaderUI.
+            local book_props = {}
+            for k, v in pairs(doc_props) do
+                book_props[k] = v
+            end
+            book_props.pages = self.ui.doc_settings:readSetting("doc_pages")
+            self:show(self.document.file, book_props)
+        end,
+    }
+end
+
+function BookInfo:isSupported(file)
+    local file_mode = lfs.attributes(file, "mode")
+    if file_mode ~= "file" then
+        return false
+    end
```

Comment:
Indeed, I forgot to return true :| The strange thing is that it still works like I expected it to work, and it shouldn't now that you pointed that to me :) I'll investigate :)
Someone indeed used `is_pic` in the original code, because calling `document:getProps()` on a `picdocument` just throw an error (and make the reader crash if you don't catch it).
As it's a bug in picdocument (ffi/C part) and should probably be fixed in another PR, I decided there and elsewhere to use ``pcall(document.getProps, document)`` to catch the case when it's a picdocument, so when pidocument is fixed, it won't need a change here. Also, picdocument could be made to return JPEG images's comment as Description, so we can save/display it.